File: tcscodevita/x.java
package tcscodevita;

import java.util.*;

public class x {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int N = input.nextInt();
        input.nextLine(); 
        String[] substrings = input.nextLine().split(" ");
        String main= input.nextLine();

        int result = removestr(main, substrings);
        System.out.println(result);
        input.close();
    }
    public static int removestr(String main , String[] substrings){
        int c=1;
        for(String i : substrings){
            if(main.contains(i)){
                main=main.replace(i,"");
                c++;
            }
        }
        return c;
    }
}

File: tcscodevita/x.py
def max_removals(main_string, substrings):
    max_count = 0  
    queue = [(main_string, 0)]  
    visited = set()  

    while queue:
        current_string, count = queue.pop(0)
        max_count = max(max_count, count)

        for substring in substrings:
            if substring in current_string:
                new_string = current_string.replace(substring, "",1)
            
                if new_string not in visited:
                    visited.add(new_string)
                    queue.append((new_string, count + 1))
                
    return max_count
n = int(input().strip())  
substrings = input().strip().split()  
main_string = input().strip()  
result = max_removals(main_string, substrings)
print(result)
